internal/redirects/redirect_store.go

```diff
@@ -0,0 +1,209 @@
+package redirects
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	netUrl "net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/SwissDataScienceCenter/renku-gateway/internal/config"
+	"github.com/labstack/echo/v4"
+)
+
+type PlatformRedirectConfig struct {
+	SourceUrl string `json:"source_url"`
+	TargetUrl string `json:"target_url"`
+}
+
+type RedirectStoreRedirectEntry struct {
+	SourceUrl string
+	TargetUrl *string
+	UpdatedAt time.Time
+}
+
+type RedirectStore struct {
+	Config      config.RedirectsStoreConfig
+	EntryTtl    time.Duration
+	RedirectMap map[string]RedirectStoreRedirectEntry
+
+	PathPrefix string
+
+	redirectedHost   string
+	redirectMapMutex sync.Mutex
+}
+
+type ServerCredentials struct {
+	Host string
+}
+
+type RedirectStoreOption func(*RedirectStore) error
+
+func WithConfig(cfg config.RedirectsStoreConfig) RedirectStoreOption {
+	return func(rs *RedirectStore) error {
+		rs.Config = cfg
+		return nil
+	}
+}
+
+func WithEntryTtl(ttl time.Duration) RedirectStoreOption {
+	return func(rs *RedirectStore) error {
+		rs.EntryTtl = ttl
+		return nil
+	}
+}
+
+func queryRenkuApi(renkuCredentials ServerCredentials, endpoint string) []byte {
+	method := "GET"
+
+	req, err := http.NewRequest(method, fmt.Sprintf("https://%s/api/data%s", renkuCredentials.Host, endpoint), nil)
```

Comment:
question: I think we have to urlencode `endpoint` right? Or is that done somewhere else?